### ISSUE TITLE
Support YAML config files

### ISSUE THREAD

**Issue by nwalters512** [PrairieLearn/PrairieLearn#1158]
It would be great to transparently support YAML course/question/etc config files for all of PL. This would be relatively easy to do, we can just add a layer of abstraction that takes a file name like `question` and resolves it to `question.yml` or `question.json` depending on which one exists and parses it with the appropriate parser.

**Comment by JamesGiaquinto** [comment]
Hi, yaml was functioning up till the recent update where it seems the yaml module is removed. I really like being able to save dictionary data in yaml, very clean format. Most all my questions use yaml that way.
Will yaml be available? Or should I find a fix on my end?

**Comment by nwalters512** [comment]
@JamesGiaquinto this issues is about supporting YAML config files (e.g. `infoCourse.yaml` instead of `infoCourse.json`), not about making the Python `yaml` module available to questions. Please open a new issue to request that.

**Comment by reteps** [comment]
I don't think we should do this IMO. As we move to GUI editors, the underlying data format becomes less important.
